src/pm/hydra/pm/pmiserv/common.c

```diff
@@ -122,11 +122,27 @@ char *HYD_pmcd_pmi_find_token_keyval(struct HYD_pmcd_token *tokens, int count, c
 HYD_status HYD_pmcd_pmi_allocate_kvs(struct HYD_pmcd_pmi_kvs ** kvs, int pgid)
 {
     HYD_status status = HYD_SUCCESS;
+    char hostname[MAX_HOSTNAME_LEN];
+    int  rnd;
+    struct timeval tv;
+    struct tm *ptm;
+    unsigned int msec;
 
     HYDU_FUNC_ENTER();
 
+    gethostname(hostname, MAX_HOSTNAME_LEN);
+    
+    gettimeofday(&tv, NULL);
+    ptm = localtime(&tv.tv_sec);
+    msec = (unsigned int)ptm->tm_hour * 3600000 +
+           (unsigned int)ptm->tm_min  * 60000   +
+           (unsigned int)ptm->tm_sec  * 1000    +
+           (unsigned int)(tv.tv_usec  / 1000);
+    srand(msec);
+    rnd = rand();
```

Comment:
Might I suggest using `srandom_r` and `random_r` instead of `srand` and `rand` here so these don't mess with the user's random number generator? They're also multithreading safe. We use them in `ch4r_symheap.h`.

Reply:
@wesbland thank you for the hint.

Reply:
@wesbland Those aren't available on OSX. Also, this is happening in `mpiexec` so I'm not sure there is cause for concern about the user here.

Reply:
Oh ok. I saw pmi and missed the pmiserv. I thought this was on the client side. 👍

Reply:
Ok so it seems rand() and srand() would work just fine for this particular case. Any comment on the seed generation? There are several options available (two of which follow):
1) microseconds in second: there would be P(collision) = 10^(-12) at any second of the day
2) milliseconds in day: there would be P(collision) = 1/(24*3600000)^2 = 1.34 * 10^(-16) at any day. This would reduce depending on the distance two mpiexecs are launched within the same day (e.g., in the same second P(collision) would only be 10^(-6)); thus microseconds in minute (2.7 * 10(-16)) would probably be better considering that mpiexecs will be unlikely to be launched in the same second and maybe not even in the same minute.

We also had some discussion about using the MAC address of the interface as seed but maybe time would be more than enough.

Reply:
IMO, MAC would be better if it's not too much trouble vs time.